src/mixins/Projector.ts

```diff
@@ -222,6 +261,13 @@ export function ProjectorMixin<P, T extends Constructor<WidgetBase<P>>>(base: T)
 			super.__setProperties__(properties);
 		}
 
+		public toHtml(): string {
+			if (this.projectorState !== ProjectorAttachState.Attached) {
+				throw new Error('Projector is not attached, cannot return an HTML string of projection.');
+			}
+			return this._projection!.domNode.outerHTML;
```

Comment:
We try not to use `!` to skip possible null checks in `src` code, could we check that there is a projection in the catch above?

```ts
if (!this._projection || this.projectorState !== ProjectorAttachState.Attached) {
```